Revit_Core_Engine/Modify/Checkout.cs

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Modifies CheckoutStatus for each element selected if element is not currently owned by the current user or others.")]
+        [Input("elements", "Revit elements to query for its checkout status.")]
+        [Output("elementsCheckedOut", "List of elements checked out (ownership modified to be by current user) by this process.")]
+        public static List<ElementId> Checkout(this List<Element> elements)
```

Comment:
Wouldn't it be better to return the actual elements, not their Ids? `elementsToCheckout` could be a `List<Element>` with `Select(x=> x.Document)` and `Select(x=> x.Id)` as arguments passed to `CheckoutElements`

Reply:
I think you're right that it would better to return the element, since in the implementation, I have to do a linq query select over the list of element to get a list of elementid to compare against this output. Simplifies this here and in all implementations.